app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java

```diff
@@ -53,26 +51,29 @@ public void setViewModel(LoginContract.ViewModel viewModel) {
     }
 
     @Override
-    public void login(String userName, String passWord) {
+    public void login(final String userName, String passWord) {
         validateUserNameInput(userName);
-        Disposable subscription = mUserRepository.login(userName, passWord)
-                .subscribeOn(mSchedulerProvider.io())
-                .observeOn(mSchedulerProvider.ui())
-                .subscribe(new Consumer<BaseResponse<SignInDataResponse>>() {
+        mUserRepository.login(userName, passWord).subscribeOn(mSchedulerProvider.io())
+                // TODO: Edit later
+                .flatMap(new Function<SignInDataResponse, Observable<SignInDataResponse>>() {
                     @Override
-                    public void accept(@NonNull BaseResponse<SignInDataResponse> signInResponse)
-                            throws Exception {
-                        mUserRepository.saveUser(signInResponse.getData().getUser());
-                        mTokenRepository.saveToken(signInResponse.getData().getAuthenToken());
-                        mViewModel.onLoginSuccess();
+                    public Observable<SignInDataResponse> apply(
+                            @NonNull SignInDataResponse signInDataResponse) throws Exception {
+                        mUserRepository.saveUser(signInDataResponse.getUser());
+                        mTokenRepository.saveToken(signInDataResponse.getAuthenToken());
+                        return Observable.just(signInDataResponse);
                     }
-                }, new RequestError() {
-                    @Override
-                    public void onRequestError(BaseException error) {
-                        mViewModel.onLoginError(error);
-                    }
-                });
-        mCompositeDisposable.add(subscription);
+                }).observeOn(mSchedulerProvider.ui()).subscribe(new Consumer<SignInDataResponse>() {
+            @Override
+            public void accept(@NonNull SignInDataResponse userProfileResponse) throws Exception {
+                mViewModel.onLoginSuccess();
+            }
```

Comment:
ERROR: 'method def rcurly' have incorrect indentation level 12, expected level should be one of the following: 20, 24.